### PR TITLE
Visualization unit tests

### LINKED ISSUE #103: Add default visualization directory
Currently, any `gender_analysis.analysis.metadata_visualization` function outputs .png files directly into the gender_analysis directory. This makes it hard for users to find and is also cluttered. Let's instead give a default output path that's either within the package or in the user's Pictures folder, if the latter is possible.

### PR DESCRIPTION
**Pull request by wilke0818**
These tests cover creating metadata-visualizations and usually check that the the files are successfully created with the correct filename and put into the correct directory. The changes in this PR also incorporate changes from #125 but has been structured to more generally create the output directories now and has moved the functionality for this into a common file.

### CODE REVIEW

**Comment by wilke0818** [comment]
@ryaanahmed fixed the issues with the unit tests that blocked it from passing on Github

File: gender_analysis/common.py [hunk 1]
```diff
@@ -320,14 +323,41 @@ def __str__(self):
...
             + self.message + ('\n' if self.message else '')
             + 'In order to run this function, you must create a new metadata csv\n'
-            + 'with ' + ('these ' if is_plural else 'this ') + 'field' + ('s ' if is_plural else ' ')
+            + 'with ' + ('these ' if is_plural else 'this ') + 'field' + (
+                's ' if is_plural else ' ')
             + 'and run Corpus.update_metadata().'
         )
 
+
+def create_path_object_and_directories(output_dir, filename=None):
```
Comment (ryaanahmed) [inline gender_analysis/common.py:335]: nice utility function!

**Review by ryaanahmed — CHANGES_REQUESTED** [review]
Looking really good. Requesting a few changes for readability.

Plz see my latest commit for how to use the `/` infix operator to construct paths. We don't need a hybrid of pathlib + os most of the time: pathlib has aliases for almost all of the os.path stuff, so better to just pick one and stick with it. (see https://docs.python.org/3/library/pathlib.html#correspondence-to-tools-in-the-os-module)

See particularly my changes to `test_create_path_object_and_directories` in the latest commit for an example of how to clean these up a little bit. If you could go through the rest of the `metadata_visualization_test` tests and similarly clean up, this will be in great shape.

**Comment by wilke0818** [comment]
@ryaanahmed Made the requested changes!

**Review by ryaanahmed — APPROVED** [review]

**Comment by ryaanahmed** [comment]
Changes look good @wilke0818 - much clearer / less duplication.

Linter failures here are all in files heavily edited in #143 so leaving alone per discussion in #145 to avoid giant merge conflicts.

merging!

**Pull request merged**